cpp/refactor_module/algorithm/refactor.cpp

```diff
@@ -474,3 +474,35 @@ void Refactor::CollapseNode(mgp_list *args, mgp_graph *memgraph_graph, mgp_resul
     return;
   }
 }
+
+void Refactor::RenameTypeProperty(mgp_list *args, mgp_graph *memgraph_graph, mgp_result *result, mgp_memory *memory) {
+  mgp::MemoryDispatcherGuard guard{memory};
+  const auto arguments = mgp::List(args);
+  const auto record_factory = mgp::RecordFactory(result);
+  try {
+    const std::string old_name{arguments[0].ValueString()};
+    const std::string new_name{arguments[1].ValueString()};
+    const auto rels = arguments[2].ValueList();
+
+    int64_t rels_changed{0};
+    for (auto rel_value : rels) {
+      auto rel = rel_value.ValueRelationship();
+      const auto prop_value = rel.GetProperty(old_name);
+      if (prop_value.IsNull()) {  // unlike mgp::Maps, property maps cant have null, so the map.merge bug shoulsnt be
+                                  // reproduced
+        continue;  // since there is no bug, it is faster to just check isNull instead of copying entire properties map
+                   // and then find
+      }  // if someone manages to reproduce map.merge bug in this function, we will change the code
```

Comment:
Thanks for the comment, but I would leave only part where you mention that mgp::Maps can have null whereas property maps can't have null, so it is faster to check if there is null and leave out part where we mention map.merge bug because others will not know what do we reference to

Reply:
And usually, comments are above the line you are commenting on. We have a little bit of everything in MAGE, but if you check Memgraph it is above the line you are commenting and it seems like good practice due to incorporating line length.